### PR TITLE
kubelet/rkt: update image related interfaces.

### PR DESCRIPTION
**Pull request by yifan-gu**
Add RemoveImage(), ListImages().
Update PullImage(), IsImagePresent().

### CODE REVIEW

**Comment by yifan-gu** [comment]
Another split from #8951 mostly update the image related interfaces
/cc @dchen1107 @yujuhong

**Comment by yujuhong** [comment]
@yifan-gu, I am not getting emails from github correctly ever since the repository was moved. Will look at this today.

**Comment by yifan-gu** [comment]
@yujuhong Thanks, btw we are planning to implement API endpoints for rkt in 0.9 release. This will help us to be more robust and efficient in getting pod status, etc.
/cc @dchen1107 ^

**Comment by jonboulle** [comment]
LGTM

File: pkg/kubelet/rkt/rkt.go [hunk 1]
```diff
@@ -800,39 +818,47 @@ func (r *runtime) PullImage(image kubecontainer.ImageSpec, pullSecrets []api.Sec
...
 
-// IsImagePresent returns true if the image is available on the machine.
-// TODO(yifan): 'rkt image' is now landed on master, use that once we bump up
-// the rkt version.
+// TODO(yifan): Searching the image via 'rkt images' might not be the most efficient way.
 func (r *runtime) IsImagePresent(image kubecontainer.ImageSpec) (bool, error) {
-	img := image.Image
-	if _, err := r.runCommand("prepare", "--local=true", dockerPrefix+img); err != nil {
-		return false, nil
+	repoToPull, tag := parsers.ParseRepositoryTag(image.Image)
```
Comment (yujuhong) [inline pkg/kubelet/rkt/rkt.go]: line 830-833 appears 4 times in this PR. Is it worth making it a function?

**Comment by yujuhong** [comment]
@k8s-bot ok to test

**Comment by yujuhong** [comment]
LGTM with one minor comment.

**Comment by yifan-gu** [comment]
@yujuhong Comments addressed.

**Pull request merged**